src/useGet.test.tsx

```diff
@@ -455,6 +455,28 @@ describe("useGet hook", () => {
     });
   });
 
+  describe("with skip", () => {
+    it("should not fetch on mount", async () => {
+      const children = jest.fn();
+      children.mockReturnValue(<div />);
+
+      const MyAwesomeComponent = () => {
+        const { data, error, loading } = useGet<{ oh: string }>({ path: "/", skip: true });
+
+        return children({ data, error, loading });
+      };
+
+      render(
+        <RestfulProvider base="https://my-awesome-api.fake">
+          <MyAwesomeComponent />
+        </RestfulProvider>,
+      );
+
+      await wait(() => expect(children).toBeCalledTimes(1));
+      expect(children).toHaveBeenCalledWith({ data: null, error: null, loading: false });
```

Comment:
I think your test doesn't test the correct thing here, nothing assert that fetch is not called here.

Can you try to revert the following change in `useGet.tsx`

```tsx
useDeepCompareEffect(() => {
     if (!props.lazy /** && !props.skip **/) {
      fetchData(props, state, setState, context, abortController);
    }
}
```

and see if your test is really catching the implementation 😉

Reply:
This test follows the pattern of asserting the child rendered 1 time (as opposed to two, which would happen if there was a network request), which I copied from the `lazy` test. I agree  the naming of both test cases is slightly misleading, but I assume it was done this way because it was hard to assert nock *wasn't* called.

Do you know if there is a good pattern for that? Happy to add it to both tests if so

Reply:
I will try to give a try locally today, I just want to be sure that we are testing the right thing (and this will not break without any red on CI 😁)